lib/ipam/ipam_block.go

```diff
@@ -53,6 +53,33 @@ func newBlock(cidr cnet.IPNet) allocationBlock {
 		b.Unallocated[i] = i
 	}
 
+	if rsvdAttr != nil {
+		// Reserve IPs based on host reserved attributes.
+		// For example, with windows OS, the following IP addresses of the block are
+		// reserved. This is done by pre-allocating them during initialization
+		// time only.
+		// IPs : x.0, x.1, x.2 and x.bcastAddr (e.g. x.255 for /24 subnet)
+
+		log.Infof("Block %s reserving IPs", b.CIDR.String())
```

Comment:
I think we use `WithFields` elsewhere to add the block to the log. Would be good to do that here as well in a consistent way so its easy to search through all the logs pertaining to a block. 

e.g., https://github.com/projectcalico/libcalico-go/blob/master/lib/ipam/ipam.go#L807